tests/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImplTest.java

```diff
@@ -401,4 +408,42 @@ private String getTestLogContents(Class testClass, String extension) {
       throw new RuntimeException(e);
     }
   }
+
+  private <T> Matcher<Iterable<? super T>> hasExactlyOneOf(final Matcher<? super T> elemMatcher) {
+    return new ExactlyOneOf<T>(elemMatcher);
+  }
+
+  private static class ExactlyOneOf<T> extends TypeSafeDiagnosingMatcher<Iterable<? super T>> {
+    private final Matcher<? super T> elemMatcher;
+
+    public ExactlyOneOf(Matcher<? super T> elemMatcher) {
+      this.elemMatcher = elemMatcher;
+    }
+
+    @Override
+    protected boolean matchesSafely(Iterable<? super T> collection,
+                                    Description mismatchDescription) {
+      boolean foundOne = false;
```

Comment:
This loop seems pretty low level. There are guava imports above, can you just make an ImmutableSet from the collection and test it's empty or has 1 element?